app/modules/map/components/location_search_input.svelte

```diff
@@ -0,0 +1,117 @@
+<script>
+  import Spinner from '#components/spinner.svelte'
+  import Flash from '#lib/components/flash.svelte'
+  import { searchLocationByText } from '#map/lib/nominatim'
+  import { i18n } from '#user/lib/i18n'
+  import { createEventDispatcher } from 'svelte'
+  import { debounce } from 'underscore'
+
+  export let inputLabel = null
+  export let inputPlaceholder = null
+
+  let results, flash
+  let lastSearchQuery = ''
+
+  let searching
+  async function searchLocation (e) {
+    const searchQuery = e.target.value
+    if (searchQuery === lastSearchQuery && results != null) return
+    lastSearchQuery = searchQuery
+    if (searchQuery === '') {
+      results = null
+      return
+    }
+    try {
+      searching = searchLocationByText(searchQuery)
+      results = await searching
+    } catch (err) {
+      flash = err
+    }
+  }
+
+  const lazySearchLocation = debounce(searchLocation, 500)
+
+  const dispatch = createEventDispatcher()
+
+  function selectResult (result) {
+    dispatch('selectLocation', result)
+    results = null
+  }
+</script>
+
+<div class="location-search-input">
+  {#if inputLabel}
+    <label for="locationSearchInput">{inputLabel}</label>
+  {/if}
+  <input
+    type="search"
+    id="locationSearchInput"
+    on:focus={lazySearchLocation}
+    on:keydown={lazySearchLocation}
+    placeholder={inputPlaceholder || i18n('Search for a location')}
+  />
+  {#await searching}
+    <div class="spinner-wrapper">
+      <Spinner />
+    </div>
+  {:then}
+    {#if results}
+      <ul tabindex="-1">
+        {#each results as result (result.osm_id)}
+          <li>
+            <button on:click={() => selectResult(result)}>
+              <span class="name">{result.name}</span>
+              <span class="description">{result.description}</span>
+            </button>
+          </li>
+        {:else}
+          <li class="no-result">{i18n('no result')}</li>
+        {/each}
+      </ul>
+    {/if}
+  {/await}
+  <Flash state={flash} />
+</div>
+
+<style lang="scss">
+  @import '#general/scss/utils';
+  input{
+    margin: 0;
+  }
+  label{
+    font-size: 1rem;
+    margin-block-end: 0.2rem;
+  }
+  ul{
+    background-color: $off-white;
+    max-height: 20em;
```

Comment:
```suggestion
    max-height: 20em;
    /* Small screens */
    @media screen and (max-width: $small-screen){
      max-height: 16em;
    }
```
to not overflow leaflet's attribution stripe

Reply:
addressed in edd45f0a8